### PR TITLE
[GG-2483]fetch state data create geojson

### PR DESCRIPTION
**Pull request by sshalu12**
fetch state data and create geojson file

### CODE REVIEW

File: script/state.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+       "geometry":state[i]
+    }
+    k.append(m)
+geo=json.dumps({
+    "type": "FeatureCollection",
+         "features": k})
+
+with open('./usr/script/state.geojson', 'w') as f:
+    f.write(geo) 
+conn.close()
```
Comment (gtvikas) [inline script/state.py]: remove EOF

**Review by gtvikas — CHANGES_REQUESTED** [review]
@sshalu12  Don't use `single letters` to name the variables. Give them proper descriptive names

File: script/state.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        dbname=database,
+        user=username,
+        password=pwd,
+        port=port_id
+    )
+cur=conn.cursor()
+
+cur.execute('SELECT * from states')
+query=cur.fetchall()
+print(query)
```
Comment (gtvikas) [inline script/state.py]: remove this print statement

File: script/state.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+#append q data in list because q fetched data in list type and element of list is tuple 
+state=[]
+for i in range(0,len(q)):
+    state.append(q[i][0])
+k=[]
+for i in range(0,len(query)):   
+    m={
+        "type":"Feature",
+         "properties": {
+                    "id":query[i][0],
```
Comment (gtvikas) [inline script/state.py]: Check the formatting here

File: script/state.py [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+                    "state":query[i][1],
+                    "country":query[i][2]
+                },
+
+       "geometry":state[i]
+    }
+    k.append(m)
+geo=json.dumps({
+    "type": "FeatureCollection",
+         "features": k})
```
Comment (gtvikas) [inline script/state.py]: Check the formatting here as well

**Review by gtvikas — CHANGES_REQUESTED** [review]

File: script/state.py [hunk 5]
```diff
@@ -0,0 +1,45 @@
+import json
+import psycopg2
+hostname = 'postgres'
+database='geovisualizer'
+username='postgres'
+pwd=''
```
Comment (gtvikas) [inline script/state.py]: What is the user of this pwd, when you are passing the blank value while creating connection

**Review by gtvikas — CHANGES_REQUESTED** [review]

File: script/state.py [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+
+port_id = 5432
+conn = psycopg2.connect(host=hostname, dbname=database, user=username, port=port_id)
+cur = conn.cursor()
+
+cur.execute("SELECT * from states")
+query = cur.fetchall()
+
+cur.execute("SELECT ST_AsGeoJSON(boundary,4326)::JSONB from states ")
+multiploygon = cur.fetchall()
```
Comment (sushilsingh95) [inline script/state.py]: Can we use a single query to all data in 1 query?

File: script/state.py [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+
+geometry = []
+state = []
+
+for i in range(0, len(query)):
```
Comment (sushilsingh95) [inline script/state.py]: can we directly iterate over query?

File: script/state.py [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+        "geometry": geometry[i],
+    }
+    state.append(feature)
+
+geo = json.dumps({"type": "FeatureCollection", "features": state})
+
+try:
+    with open("./usr/script/state.geojson", "w") as f:
+        f.write(geo)
+except EOFError as e:
```
Comment (sushilsingh95) [inline script/state.py]: Is this the only type of error that we get. E.g. can we get permissions error?

File: script/state.py [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+    print(e)
+
+conn.close()
```
Comment (sushilsingh95) [inline script/state.py]: can we move this line to just after we are done with all db queries?

File: docker-compose.yml [hunk 10]
```diff
@@ -10,7 +10,7 @@ services:
       - postgres
     volumes:
       - ./script:/usr/script
-    command: [ "./usr/script/wait-for-postgres.sh","postgres","python", "./usr/script/test.py"]
+    command: [ "./usr/script/wait-for-postgres.sh","postgres","python", "./usr/script/state.py"]
```
Comment (sushilsingh95) [inline docker-compose.yml]: is there any particular reason for creating the file in this directory

**Review by sushilsingh95 — CHANGES_REQUESTED** [review]

**Comment by ashutoshgt** [comment]
Please use docker-compose run with makefile to parametrize the command for our python service in the docker-compose file.

File: script/state.py [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+
+port_id = 5432
+conn = psycopg2.connect(host=hostname, dbname=database, user=username, port=port_id)
+cur = conn.cursor()
+
+cur.execute("SELECT id,state,country,ST_AsGeoJSON(boundary,4326)::JSONB from states")
+query = cur.fetchall()
+conn.close()
+
+state = []
```
Comment (ashutoshgt) [inline script/state.py]:
```suggestion
states = []
```

File: script/state.py [hunk 12]
```diff
@@ -0,0 +1,34 @@
...
+hostname = "postgres"
+database = "geovisualizer"
+username = "postgres"
+
+port_id = 5432
+conn = psycopg2.connect(host=hostname, dbname=database, user=username, port=port_id)
+cur = conn.cursor()
+
+cur.execute("SELECT id,state,country,ST_AsGeoJSON(boundary,4326)::JSONB from states")
+query = cur.fetchall()
```
Comment (ashutoshgt) [inline script/state.py]: These are results of the query, hence can be named better like `query_results` or `states_data`

File: script/state.py [hunk 13]
```diff
@@ -0,0 +1,34 @@
...
+conn.close()
+
+state = []
+
+for i in query:
```
Comment (ashutoshgt) [inline script/state.py]:
```suggestion
for row in states_data:
```

**Review by ashutoshgt — CHANGES_REQUESTED** [review]

**Review by ashutoshgt — APPROVED** [review]

**Pull request merged**